src/global.h

```diff
@@ -319,11 +319,41 @@ typedef unsigned char      uint8_t;
 // definition for custom event
 #define MS_PACKET_RECEIVED 0
 
-/* Classes ********************************************************************/
+/* Exception Classes **********************************************************/
+
+class CInfoExit
```

Comment:
> It's [bad style](https://isocpp.org/wiki/faq/exceptions#why-not-exceptions) in C++ to use exceptions for handling normal situations.

It's what you call a "normal" situation...
But I agree that, just for the --help and --version, it could be a return ( 0 ).